diff --recursive '--unified=3' original/boost_1_33_1/libs/regex/src/cregex.cpp modified/boost_1_33_1/libs/regex/src/cregex.cpp
--- original/boost_1_33_1/libs/regex/src/cregex.cpp	2005-03-30 13:38:32.000000000 +0200
+++ modified/boost_1_33_1/libs/regex/src/cregex.cpp	2014-12-06 23:53:30.273125000 +0100
@@ -362,7 +362,7 @@
 #if BOOST_WORKAROUND(BOOST_MSVC, >= 1400)
          (::sprintf_s)(buf, sizeof(buf), "%s%s%s", dstart.path(), directory_iterator::separator(), ptr);
 #else
-         (std::sprintf)(buf, "%s%s%s", dstart.path(), directory_iterator::separator(), ptr);
+         sprintf(buf, "%s%s%s", dstart.path(), directory_iterator::separator(), ptr);
 #endif
          BuildFileList(pl, buf, recurse);
          ++dstart;
diff --recursive '--unified=3' original/boost_1_33_1/libs/regex/src/fileiter.cpp modified/boost_1_33_1/libs/regex/src/fileiter.cpp
--- original/boost_1_33_1/libs/regex/src/fileiter.cpp	2005-03-30 13:38:32.000000000 +0200
+++ modified/boost_1_33_1/libs/regex/src/fileiter.cpp	2014-12-06 23:53:19.195000000 +0100
@@ -30,7 +30,6 @@
 #include <cstdio>
 #if defined(BOOST_NO_STDC_NAMESPACE)
 namespace std{
-   using ::sprintf;
    using ::fseek;
    using ::fread;
    using ::ftell;
@@ -788,9 +787,9 @@
 {
    char buf[MAX_PATH];
    if( ( (root[0] == *_fi_sep) || (root[0] == *_fi_sep_alt) ) && (root[1] == '\0') )
-      (std::sprintf)(buf, "%s%s", root, name);
+      sprintf(buf, "%s%s", root, name);
    else
-      (std::sprintf)(buf, "%s%s%s", root, _fi_sep, name);
+      sprintf(buf, "%s%s%s", root, _fi_sep, name);
    DIR* d = opendir(buf);
    if(d)
    {
diff --recursive '--unified=3' original/boost_1_33_1/libs/regex/src/posix_api.cpp modified/boost_1_33_1/libs/regex/src/posix_api.cpp
--- original/boost_1_33_1/libs/regex/src/posix_api.cpp	2005-03-30 13:38:32.000000000 +0200
+++ modified/boost_1_33_1/libs/regex/src/posix_api.cpp	2014-12-07 02:44:11.652125000 +0100
@@ -24,7 +24,6 @@
 
 #if defined(BOOST_NO_STDC_NAMESPACE)
 namespace std{
-   using ::sprintf;
    using ::strcpy;
    using ::strcmp;
 }
@@ -148,7 +147,7 @@
 #if BOOST_WORKAROUND(BOOST_MSVC, >= 1400)
             (::sprintf_s)(localbuf, 5, "%d", i);
 #else
-            (std::sprintf)(localbuf, "%d", i);
+            sprintf(localbuf, "%d", i);
 #endif
             if(std::strlen(localbuf) < buf_size)
                re_detail::strcpy_s(buf, buf_size, localbuf);
@@ -158,7 +157,7 @@
 #if BOOST_WORKAROUND(BOOST_MSVC, >= 1400)
       (::sprintf_s)(localbuf, 5, "%d", 0);
 #else
-      (std::sprintf)(localbuf, "%d", 0);
+      sprintf(localbuf, "%d", 0);
 #endif
       if(std::strlen(localbuf) < buf_size)
          re_detail::strcpy_s(buf, buf_size, localbuf);
