diff --recursive --unified=3 original/wxWidgets-2.8.10/include/wx/thread.h modified/wxWidgets-2.8.10/include/wx/thread.h
--- original/wxWidgets-2.8.10/include/wx/thread.h	2009-03-06 12:10:52.000000000 +0000
+++ modified/wxWidgets-2.8.10/include/wx/thread.h	2009-04-25 02:35:48.000000000 +0000
@@ -688,8 +688,8 @@
 #else // !wxUSE_THREADS
 
 // no thread support
-inline void WXDLLIMPEXP_BASE wxMutexGuiEnter() { }
-inline void WXDLLIMPEXP_BASE wxMutexGuiLeave() { }
+inline void wxMutexGuiEnter() { }
+inline void wxMutexGuiLeave() { }
 
 // macros for entering/leaving critical sections which may be used without
 // having to take them inside "#if wxUSE_THREADS"
