diff --recursive --unified=3 original/wxWidgets-2.8.6/src/msw/textctrl.cpp modified/wxWidgets-2.8.6/src/msw/textctrl.cpp
--- original/wxWidgets-2.8.6/src/msw/textctrl.cpp	2007-09-25 08:52:06.000000000 +0000
+++ modified/wxWidgets-2.8.6/src/msw/textctrl.cpp	2008-02-04 00:48:03.802769300 +0000
@@ -1964,6 +1964,19 @@
 
 WXLRESULT wxTextCtrl::MSWWindowProc(WXUINT nMsg, WXWPARAM wParam, WXLPARAM lParam)
 {
+    // we must handle clipboard events before calling MSWWindowProc, otherwise
+    // the event would be handled twice if there's a handler for it in user
+    // code:
+    switch ( nMsg )
+    {
+        case WM_CUT:
+        case WM_COPY:
+        case WM_PASTE:
+            if ( HandleClipboardEvent(nMsg) )
+                return 0;
+            break;
+    }
+
     WXLRESULT lRc = wxTextCtrlBase::MSWWindowProc(nMsg, wParam, lParam);
 
     switch ( nMsg )
@@ -2006,13 +2019,6 @@
                 }
             }
             break;
-
-        case WM_CUT:
-        case WM_COPY:
-        case WM_PASTE:
-            if ( HandleClipboardEvent(nMsg) )
-                lRc = 0;
-            break;
     }
 
     return lRc;
