diff -x 'xml*' --recursive '--unified=3' original/wxWidgets-2.9.5/include/wx/cpp.h modified/wxWidgets-2.9.5/include/wx/cpp.h
--- original/wxWidgets-2.9.5/include/wx/cpp.h	2013-07-16 15:47:14.000000000 +0000
+++ modified/wxWidgets-2.9.5/include/wx/cpp.h	2013-07-20 00:40:10.000000000 +0000
@@ -144,6 +144,13 @@
 
 
 #ifdef HAVE_VARIADIC_MACROS
+/* There is no other way to suppress the
+ *  "anonymous variadic macros were introduced in C99"
+ * diagnostic with gcc-3.x .
+ */
+#  if defined(__GNUC__) && __GNUC__ == 3
+#    pragma GCC system_header
+#  endif /* gcc-3.x */
 /*
    wxCALL_FOR_EACH(what, ...) calls the macro from its first argument, what(pos, x),
    for every remaining argument 'x', with 'pos' being its 1-based index in
