Index: wxWidgets-2.9.3/src/common/stopwatch.cpp
===================================================================
--- old/wxWidgets-2.9.3/src/common/stopwatch.cpp (revision 69921)
+++ new/wxWidgets-2.9.3/src/common/stopwatch.cpp (revision 70012)
@@ -38,4 +38,5 @@
     #endif
     #include "wx/log.h"
+    #include "wx/thread.h"
 #endif //WX_PRECOMP
 
@@ -65,5 +66,5 @@
     }
 
-    wxCriticalSection cs;
+    wxCRIT_SECT_DECLARE_MEMBER(cs);
     LARGE_INTEGER freq;
     bool init;
@@ -83,5 +84,5 @@
     if ( !gs_perfCounter.init )
     {
-        wxCriticalSectionLocker lock(gs_perfCounter.cs);
+        wxCRIT_SECT_LOCKER(lock, gs_perfCounter.cs);
         ::QueryPerformanceFrequency(&gs_perfCounter.freq);

Index: old/wxWidgets-2.9.3/src/generic/datavgen.cpp
===================================================================
--- old/wxWidgets-2.9.3/src/generic/datavgen.cpp (revision 70201)
+++ old/wxWidgets-2.9.3/src/generic/datavgen.cpp (revision 70308)
@@ -623,4 +623,7 @@
     void ChangeCurrentRow( unsigned int row );
     bool TryAdvanceCurrentColumn(wxDataViewTreeNode *node, bool forward);
+
+    wxDataViewColumn *GetCurrentColumn() const { return m_currentCol; }
+    void ClearCurrentColumn() { m_currentCol = NULL; }
 
     bool IsSingleSel() const { return !GetParent()->HasFlag(wxDV_MULTIPLE); }
@@ -4743,4 +4746,8 @@
     m_colsBestWidths.erase(m_colsBestWidths.begin() + GetColumnIndex(column));
     m_cols.Erase(ret);
+
+    if ( m_clientArea->GetCurrentColumn() == column )
+        m_clientArea->ClearCurrentColumn();
+
     OnColumnsCountChanged();
 
@@ -4753,5 +4760,9 @@
     m_cols.Clear();
     m_colsBestWidths.clear();
+
+    m_clientArea->ClearCurrentColumn();
+
     OnColumnsCountChanged();
+
     return true;
 }
Index: old/wxWidgets-2.9.3/src/generic/datavgen.cpp
===================================================================
--- old/wxWidgets-2.9.3/src/generic/datavgen.cpp (revision 70308)
+++ old/wxWidgets-2.9.3/src/generic/datavgen.cpp (revision 70318)
@@ -3528,4 +3528,5 @@
     {
         case WXK_RETURN:
+            if ( !event.HasModifiers() )
             {
                 // Enter activates the item, i.e. sends wxEVT_COMMAND_DATAVIEW_ITEM_ACTIVATED to
@@ -3547,4 +3548,5 @@

         case WXK_SPACE:
+            if ( !event.HasModifiers() )
             {
                 // Space toggles activatable items or -- if not activatable --
@@ -3576,4 +3578,5 @@

         case WXK_F2:
+            if ( !event.HasModifiers() )
             {
                 if( !m_selection.empty() )

