diff --git a/src/msw/spinctrl.cpp b/src/msw/spinctrl.cpp
index ea0f28e..8c50bfe 100644
--- a/src/msw/spinctrl.cpp
+++ b/src/msw/spinctrl.cpp
@@ -319,7 +319,7 @@ bool wxSpinCtrl::Create(wxWindow *parent,
 
 
     // create the spin button
-    if ( !wxSpinButton::Create(parent, id, wxPoint(0, 0), wxSize(0, 0), style, name) )
+    if ( !wxSpinButton::Create(parent, id, pos, wxSize(0, 0), style, name) )
     {
         return false;
     }

