diff --recursive '--unified=3' original/xmlwrapp-0.8.1/include/xmlwrapp/node.h modified/xmlwrapp-0.8.1/include/xmlwrapp/node.h
--- original/xmlwrapp-0.8.1/include/xmlwrapp/node.h	2016-01-18 14:21:44.000000000 +0000
+++ modified/xmlwrapp-0.8.1/include/xmlwrapp/node.h	2017-09-30 21:36:12.271994244 +0000
@@ -162,7 +162,7 @@
      */
     struct text
     {
-        explicit text (const char *text) : t(text) {}
+        explicit text (const char *content) : t(content) {}
         const char *t;
     };
 
diff --recursive '--unified=3' original/xmlwrapp-0.8.1/src/libxml/node.cxx modified/xmlwrapp-0.8.1/src/libxml/node.cxx
--- original/xmlwrapp-0.8.1/src/libxml/node.cxx	2016-12-11 14:28:47.000000000 +0000
+++ modified/xmlwrapp-0.8.1/src/libxml/node.cxx	2017-09-30 21:50:56.765557364 +0000
@@ -589,7 +589,7 @@
         {
             if ( ns->prefix == NULL )
             {
-                xmlFree((void*)ns->href);
+                xmlFree((void*)const_cast<xmlChar*>(ns->href));
                 ns->href = xmlStrdup(xmlHref);
                 break;
             }
