diff --recursive '--unified=3' original/xmlwrapp-0.7.0/include/xmlwrapp/export.h modified/xmlwrapp-0.7.0/include/xmlwrapp/export.h
--- original/xmlwrapp-0.7.0/include/xmlwrapp/export.h	2013-03-16 11:29:12.000000000 +0000
+++ modified/xmlwrapp-0.7.0/include/xmlwrapp/export.h	2013-04-24 13:17:36.000000000 +0000
@@ -48,7 +48,8 @@
         #define XMLWRAPP_DEPRECATED(msg) __attribute__((deprecated))
     #endif
 #elif defined(__GNUC__)
-    #define XMLWRAPP_DEPRECATED(msg) __attribute__((deprecated))
+//    #define XMLWRAPP_DEPRECATED(msg) __attribute__((deprecated))
+    #define XMLWRAPP_DEPRECATED(msg)
 #elif defined(__VISUALC__) && (__VISUALC__ >= 1300)
     #define XMLWRAPP_DEPRECATED(msg) __declspec(deprecated)
 #else
diff --recursive '--unified=3' original/xmlwrapp-0.7.0/src/libxml/errors.cxx modified/xmlwrapp-0.7.0/src/libxml/errors.cxx
--- original/xmlwrapp-0.7.0/src/libxml/errors.cxx	2013-03-16 11:29:12.000000000 +0000
+++ modified/xmlwrapp-0.7.0/src/libxml/errors.cxx	2013-04-24 13:35:38.000000000 +0000
@@ -95,6 +95,8 @@
         case error_message::type_warning:
             return "warning: " + msg.message();
     }
+
+    return "xmlwrapp internal error: " + msg.message();
 }
 
 // ------------------------------------------------------------------------
@@ -186,6 +188,8 @@
         case error_message::type_warning:
             return "warning: " + msg.message();
     }
+
+    return "xmlwrapp internal error: " + msg.message();
 }
 
 } // namespace impl
